### PR TITLE
ArC - make it possible to turn any annotation into a qualifier

### LINKED ISSUE #14822: ArC - make it possible to turn an annotation into a qualifier without adding @Qualifier to it
I.e. add an alternative to `BeforeBeanDiscovery.addQualifier()`

### PR DESCRIPTION
**Pull request by mkouba**
- resolves #14822

IMPL. NOTE: because of qualifiers are needed for resolution at runtime we have to record the qualifiers -> non-binding members map (in contrast with interceptor bindings).

### CODE REVIEW

File: docs/src/main/asciidoc/cdi-integration.adoc
Comment (manovotn) [inline docs/src/main/asciidoc/cdi-integration.adoc:456]: This needs different ID.

**Review by manovotn — COMMENTED** [review]
LGTM just one corner case question - can you use this to register a repeating qualifier?

**Comment by Ladicek** [comment]
Looks good to me. I guess the answer to the question about repeating qualifiers is: if the annotation is already `@Repeatable`, then the qualifier will also be repeatable?

**Comment by mkouba** [comment]
> Looks good to me. I guess the answer to the question about repeating qualifiers is: if the annotation is already `@Repeatable`, then the qualifier will also be repeatable?

Yes, that should work. Actually, I've added a test ;-).

**Review by Ladicek — APPROVED** [review]

**Comment by mkouba** [comment]
Let's wait for @manovotn before we merge...

**Review by manovotn — APPROVED** [review]

**Pull request merged**
